### PR TITLE
[POC][bugfix] Libraries without SONAME flag

### LINKED ISSUE #3230: [build system] Link with shared lib (linux) without SONAME enabled (from autotools)
Creating packages with shared libs that doesn't have SONAME enabled is a problem for later consuming and linking with them:

- Check if there is some possible way to force autotools to link them
- Is it possible to detect it and warn? (Will be kind of "binary linters" feature, like https://github.com/conan-io/conan/issues/1115)
- At least, if not possible to develop something, document it in the docs.

### PR DESCRIPTION
**Pull request by franramirez688**
Changelog: Bugfix: Dynamic libraries without SONAME flag.
Closes: https://github.com/conan-io/conan/issues/3230
Docs: https://github.com/conan-io/docs/pull/XXXX

- [ ] Refer to the issue that supports this Pull Request.
- [ ] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [ ] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [ ] I've followed the PEP8 style guides for Python code.
- [ ] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. 

<sup>**Note:** By default this PR will skip the slower tests and will use a limited set of python versions. Check [here](https://github.com/conan-io/conan/blob/develop/.github/PR_INCREASE_TESTING.md) how to increase the testing level by writing some tags in the current PR body text.</sup>

### CODE REVIEW

File: conan/tools/cmake/cmakedeps/templates/target_configuration.py [hunk 1]
```diff
@@ -31,7 +31,8 @@ def context(self):
                 "config_suffix": self.config_suffix,
                 "deps_targets_names": ";".join(deps_targets_names),
                 "components_names": components_names,
-                "configuration": self.cmakedeps.configuration}
+                "configuration": self.cmakedeps.configuration,
+                "nosoname": str(self.nosoname).upper()}
```
Comment (memsharded) [inline conan/tools/cmake/cmakedeps/templates/target_configuration.py:35]: I think I would try to put this as a ``xxxx-data.cmake`` variable, then use it. Like, all the parametrization of a dependency should go to ``xxx-data.cmake``, and the rest of the files use the data from there to act accordingly.

File: conans/test/functional/toolchains/cmake/test_cmake_and_no_soname_flag.py [hunk 2]
```diff
@@ -66,72 +81,89 @@ def package_info(self):
...
-    client.run_command('cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="./conan_toolchain.cmake" .'
-                       ' && cmake --build . && ./example')
+    cpp = gen_function_cpp(name="main", includes=["lib_b"], calls=["lib_b"])
+    client2.save({"CMakeLists.txt": cmakelists.format(current_folder=client.current_folder),
+                  "src/example.cpp": cpp,
+                  "conanfile.txt": conanfile},
+                 clean_first=True)
+    client2.run('install . ')
+    # Activate the VirtualRunEnv and execute the CMakeToolchain
+    client2.run_command('bash -c \'source conanrun.sh '
```
Comment (memsharded) [inline conans/test/functional/toolchains/cmake/test_cmake_and_no_soname_flag.py:167]: why need to launch ``bash`` here? this should work without it?

**Review by memsharded — COMMENTED** [review]
No need to change anything at the moment, just 2 comments, lets wait until the shared/static thing is a bit more clear.

File: conans/test/functional/toolchains/cmake/test_cmake_and_no_soname_flag.py (hunk 2 above)
Comment (franramirez688) [reply conans/test/functional/toolchains/cmake/test_cmake_and_no_soname_flag.py:167]: Yeah, I don't get to remember it... Perhaps it was due to trying to simulate a complete scenario without Conan executing the toolchain but, yes, I think we could run a `conan create`, and that's all.

**Comment by franramirez688** [comment]
Just keeping in mind that this POC is related to this another one https://github.com/conan-io/conan/pull/10997
It's strongly recommended that we should have solved the problem of classifying `shared`/`static` libraries before.

**Comment by franramirez688** [comment]
Superseded by https://github.com/conan-io/conan/pull/11318

**Pull request closed without merging**